haystack/document_store/elasticsearch.py

```diff
@@ -387,10 +392,12 @@ def get_label_count(self, index: Optional[str] = None) -> int:
         return self.get_document_count(index=index)
 
     def get_all_documents(
```

Comment:
With the new params, this `fn` can be configured to return less than all documents making the name of this fn quite confusing. Probably not somethign to deal with in this PR (not least since it would be a breaking change) but definitely worth addressing as an issue.